### PR TITLE
DEV-190: Create CIS Minor

### PR DESCRIPTION
**Pull request by D3Rkness**
Description
CIS minor & requirements does not exist.

Implementation
Major is created, added requirements to cis minor.

Testing
Tested locally and everything works as expected.

### CODE REVIEW

**Comment by D3Rkness** [comment]
May need a check on "design course in CIS." I think I've included most of the design courses (referenced other majors that requires design courses), but I could not find a full, concrete list that lists out all allowed design courses for CIS.

**Review by mattliu-mygit — APPROVED** [review]
Nice!

**Review by aryamanshodhan — APPROVED** [review]
great job!

File: lib/resources/majors.tsx [hunk 1]
```diff
@@ -5077,6 +5330,180 @@ const minorMath: Minor = {
...
+        'Math requirements may also be satisfied by an equivalent course as determined by CIS advisor.',
+      criteria:
+        'AS.110.106[C]^OR^AS.110.108[C]^OR^AS.110.109[C]^OR^AS.110.107[C]^OR^' +
+        'AS.110.202[C]^OR^AS.110.211[C]^OR^AS.110.201[C]^OR^EN.553.291[C]^OR^AS.110.212[C]',
+      double_count: [
+        'Humanities (H) Distribution',
+        'Social Science (S) Distribution',
+        'Other (N/E/Q) Distribution',
+        'Writing Intensive',
+      ],
```
Comment (mpark63) [inline lib/resources/majors.tsx:5379]: not needed, these are not distributions in the minor

File: lib/resources/majors.tsx [hunk 2]
```diff
@@ -5077,6 +5330,180 @@ const minorMath: Minor = {
...
+      required_credits: 8,
+      min_credits_per_course: 4,
+      description:
+        'You must take Computer Integrated Surgery I and a design course in CIS.',
+      criteria:
+        '(EN.510.433[C]^OR^EN.510.434[C])^OR^(EN.520.462[C]^OR^EN.520.463[C])^OR^' +
+        '(EN.520.498[C]^OR^EN.520.499[C])^OR^(EN.540.400[C]^OR^EN.540.421[C])^OR^' +
+        '(EN.580.311[C]^OR^EN.580.312[C])^OR^(EN.580.411[C]^OR^EN.580.412[C])^OR^' +
+        '(EN.580.456[C]^OR^EN.580.457[C])^OR^(EN.580.471[C]^OR^EN.580.571[C])^OR^' +
+        '(EN.580.480[C]^OR^EN.580.481[C])^OR^(EN.580.580[C]^OR^EN.580.581[C])^OR^' +
```
Comment (mpark63) [inline lib/resources/majors.tsx:5427]: why the pairwise parenthesis?

File: lib/resources/majors.tsx [hunk 3]
```diff
@@ -5077,6 +5330,180 @@ const minorMath: Minor = {
...
+          description:
+            '<b>EN.601.456 Computer Integrated Surgery II or design course in CIS</b>',
+          required_credits: 4,
+          criteria:
+            '(EN.510.433[C]^OR^EN.510.434[C])^OR^(EN.520.462[C]^OR^EN.520.463[C])^OR^' +
+            '(EN.520.498[C]^OR^EN.520.499[C])^OR^(EN.540.400[C]^OR^EN.540.421[C])^OR^' +
+            '(EN.580.311[C]^OR^EN.580.312[C])^OR^(EN.580.411[C]^OR^EN.580.412[C])^OR^' +
+            '(EN.580.456[C]^OR^EN.580.457[C])^OR^(EN.580.471[C]^OR^EN.580.571[C])^OR^' +
+            '(EN.580.480[C]^OR^EN.580.481[C])^OR^(EN.580.580[C]^OR^EN.580.581[C])^OR^' +
+            'EN.601.456[C]^OR^(EN.580.437[C]^OR^EN.580.438[C])',
```
Comment (mpark63) [inline lib/resources/majors.tsx:5445]: do these include design courses not in CIS? uwu

A design course in CIS. Either EN.601.456 Computer Integrated Surgery II or a design course in biomedical engineering, electrical and computer engineering, or mechanical engineering with substantial CIS content approved by the student’s faculty advisor in the CIS minor.

Comment (mpark63) [reply lib/resources/majors.tsx:5445]: also is there a better way to query this like 

if design is a tag you could do design tag + department

File: lib/resources/majors.tsx [hunk 4]
```diff
@@ -5077,6 +5330,180 @@ const minorMath: Minor = {
...
+            '<br /> <i>OR</i> <br />' +
+            'EN.601.463 Algorithms for Sensor-Based Robotics',
+          criteria:
+            'EN.520.414[C]^OR^EN.520.432[C]^OR^EN.520.433[C]^OR^EN.601.461[C]^OR^EN.530.420[C]^OR^EN.530.421[C]^OR^EN.530.603[C]^OR^EN.530.646[C]^OR^EN.601.463[C]',
+          double_count: ['All'],
+        },
+        {
+          required_credits: 0,
+          description:
+            '<b>Other</b> <br />' +
```
Comment (mpark63) [inline lib/resources/majors.tsx:5486]: clever implementation!

File: lib/resources/majors.tsx [hunk 5]
```diff
@@ -5122,6 +5549,7 @@ export const allMajors: Major[] = [
   baMath,
   baPhysics,
   bsPhysics,
+  minorCIS,
   // bsEnvEng,
   // baSoc,
```
Comment (mpark63) [inline lib/resources/majors.tsx:5554]: hehe nitpick but can you put all the majors together and minors together?

**Review by mpark63 — APPROVED** [review]

**Comment by D3Rkness** [comment]
DEV-136 Create BA Chem Major (link: https://github.com/uCredit-Dev/ucredit_frontend_typescript/pull/484) is included in DEV-190 Create CIS Minor. Thus, DEV-136 was closed and DEV-190 contains the updated code with both the CIS Minor implementation as well as the BA Chem implementation.

**Pull request merged**